source/common/router/router.cc

```diff
@@ -1185,6 +1185,8 @@ void Filter::onSoftPerTryTimeout(UpstreamRequest& upstream_request) {
   }
 }
 
+void Filter::onHeadersTransmittedToUpstream() { upstream_request_started_ = true; }
```

Comment:
rather than adding yet another API and  yet another boolean, can we check upstream requests and see if any have the first upstream tx byte sent in their upstream timing?
it's a bit annoying as we'd potentially have to check multiple upstream requests, in the case of hedging, but with n=2 I think it's doable, WDYT?

Reply:
I think I follow; so the data model is that each router instance has a set of upstream requests (size is <= 2 due to hedging) and we need to see if any of them have transmitted headers upstream